source/simulator/newton.cc

```diff
@@ -161,6 +161,239 @@ namespace aspect
+          return "SPD";
+          break;
+        case PreconditionerStabilization::PD:
+          return "PD";
+          break;
+        case PreconditionerStabilization::symmetric:
+          return "symmetric";
+          break;
+        case PreconditionerStabilization::none:
+          return "none";
+          break;
+      }
+  }
+
+  template <int dim>
+  std::string
+  NewtonHandler<dim>::
+  get_velocity_block_stabilization_string(VelocityBlockStabilization velocity_block_stabilization)
+  {
+    switch (velocity_block_stabilization)
+      {
+        case VelocityBlockStabilization::SPD:
+          return "SPD";
+          break;
+        case VelocityBlockStabilization::PD:
+          return "PD";
+          break;
+        case VelocityBlockStabilization::symmetric:
+          return "symmetric";
+          break;
+        case VelocityBlockStabilization::none:
+          return "none";
+          break;
+      }
+  }
+
+  template <int dim>
+  void
+  NewtonHandler<dim>::
+  declare_parameters (ParameterHandler &prm)
+  {
+    prm.enter_subsection ("Solver parameters");
+    {
+      prm.enter_subsection ("Newton solver parameters");
+      {
+        prm.declare_entry ("Nonlinear Newton solver switch tolerance", "1e-5",
+                           Patterns::Double(0,1),
+                           "A relative tolerance with respect to the residual of the first "
+                           "iteration, up to which the nonlinear Picard solver will iterate, "
+                           "before changing to the newton solver.");
+
+        prm.declare_entry ("Max pre-Newton nonlinear iterations", "10",
+                           Patterns::Integer (0),
+                           "The maximum number of Picard nonlinear iterations to be performed "
+                           "before switching to Newton iterations.");
+
+        prm.declare_entry ("Max Newton line search iterations", "5",
+                           Patterns::Integer (0),
+                           "The maximum number of line search iterations allowed. If the "
+                           "criterion is not reached after this iteration, we apply the scaled "
+                           "increment to the solution and continue.");
+
+        prm.declare_entry ("Use Newton residual scaling method", "false",
+                           Patterns::Bool (),
+                           "This method allows to slowly introduce the derivatives based on the improvement "
+                           "of the residual. If set to false, the scaling factor for the Newton derivatives "
+                           "is set to one immediately when switching on the Newton solver.");
+
+        prm.declare_entry ("Maximum linear Stokes solver tolerance", "0.9",
+                           Patterns::Double (0,1),
+                           "When the linear Stokes solver tolerance is dynamically chosen, this defines "
+                           "the most loose tolerance allowed.");
```

Comment:
I don't know what this parameter does, either from the parameter's name nor from the description. Can you elaborate?